Using cache found in C:\Users\mhroh/.cache\torch\hub\facebookresearch_dinov2_main
C:\Users\mhroh/.cache\torch\hub\facebookresearch_dinov2_main\dinov2\layers\swiglu_ffn.py:51: UserWarning: xFormers is not available (SwiGLU)
  warnings.warn("xFormers is not available (SwiGLU)")
C:\Users\mhroh/.cache\torch\hub\facebookresearch_dinov2_main\dinov2\layers\attention.py:33: UserWarning: xFormers is not available (Attention)
  warnings.warn("xFormers is not available (Attention)")
C:\Users\mhroh/.cache\torch\hub\facebookresearch_dinov2_main\dinov2\layers\block.py:40: UserWarning: xFormers is not available (Block)
  warnings.warn("xFormers is not available (Block)")
Using cache found in C:\Users\mhroh/.cache\torch\hub\facebookresearch_dinov2_main
Total parameters: 116806593
C:\Users\mhroh\Depth_Anything\src\train.py:171: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = GradScaler()
0it [00:00, ?it/s]C:\Users\mhroh\Depth_Anything\src\train.py:185: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
2
  with autocast(dtype=torch.bfloat16):
torch.Size([2, 1024, 768])
gt ( 1st element ): tensor([0.5701, 0.5701, 0.5701,  ..., 0.8224, 0.8224, 0.8224], device='cuda:0')
pred ( 1st element ): tensor([0.0356, 0.0342, 0.0371,  ..., 0.0525, 0.0544, 0.0415], device='cuda:0',
       grad_fn=<SelectBackward0>)
y aggregation ( 1st element ): tensor(167838.7812, device='cuda:0')
pred aggregation ( 1st element ): tensor(8624.0908, device='cuda:0', grad_fn=<SumBackward1>)
feature_loss skipped!!
Epoch [1/20], Batch [0], Loss: 3.4762
1it [00:10, 10.66s/it]
Epoch [1/20] Loss: 3.4762
2
C:\Users\mhroh\Depth_Anything\src\train.py:211: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(dtype=torch.bfloat16):
torch.Size([2, 1024, 768])
gt ( 1st element ): tensor([0.0000, 0.0000, 0.0000,  ..., 0.8297, 0.8341, 0.8428], device='cuda:0')
pred ( 1st element ): tensor([0.0342, 0.0356, 0.0381,  ..., 0.0527, 0.0562, 0.0388], device='cuda:0')
y aggregation ( 1st element ): tensor(129936.6094, device='cuda:0')
pred aggregation ( 1st element ): tensor(9542.7998, device='cuda:0')
feature_loss skipped!!
Traceback (most recent call last):
  File "C:\Users\mhroh\anaconda3\envs\dl_env\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\mhroh\anaconda3\envs\dl_env\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\mhroh\Depth_Anything\src\train.py", line 287, in <module>
    main()
  File "C:\Users\mhroh\Depth_Anything\src\train.py", line 284, in main
    train_student()
  File "C:\Users\mhroh\Depth_Anything\src\train.py", line 215, in train_student
    outputs = scale_shift_correction(outputs, targets)
  File "C:\Users\mhroh\Depth_Anything\src\utils\metrics.py", line 12, in scale_shift_correction
    h_opt = torch.inverse(element_1) @ element_2
RuntimeError: linalg.inv: Low precision dtypes not supported. Got BFloat16
Traceback (most recent call last):
  File "C:\Users\mhroh\anaconda3\envs\dl_env\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\mhroh\anaconda3\envs\dl_env\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\mhroh\Depth_Anything\src\train.py", line 287, in <module>
    main()
  File "C:\Users\mhroh\Depth_Anything\src\train.py", line 284, in main
    train_student()
  File "C:\Users\mhroh\Depth_Anything\src\train.py", line 215, in train_student
    outputs = scale_shift_correction(outputs, targets)
  File "C:\Users\mhroh\Depth_Anything\src\utils\metrics.py", line 12, in scale_shift_correction
    h_opt = torch.inverse(element_1) @ element_2
RuntimeError: linalg.inv: Low precision dtypes not supported. Got BFloat16
